src/layers/utils.js

```diff
@@ -0,0 +1,71 @@
+export const MAX_COLOR_INTENSITY = 255;
+
+export const DEFAULT_COLOR_OFF = [0, 0, 0];
+
+export const MAX_SLIDERS_AND_CHANNELS = 6;
+
+export function range(len) {
+  return [...Array(len).keys()];
+}
+
+export function padWithDefault(arr, defaultValue, padWidth) {
+  for (let i = 0; i < padWidth; i += 1) {
+    arr.push(defaultValue);
+  }
+  return arr;
+}
+
+export function padColorsAndSliders({
```

Comment:
I renamed this and restructured this function and changed its arguments as it was causing problems in testing.  Passing down props and destructuring them isn't so simple in deck.gl because the Object has other methods (as I have discovered).  Thus it's better to take what you need only.